Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/connected_components.h

```diff
@@ -232,10 +232,15 @@ connected_components(const PolygonMesh& pmesh,
   FiniteDual finite_dual(dual,
     internal::No_border<PolygonMesh, EdgeConstraintMap>(pmesh, ecmap));
 
-  typename GetFaceIndexMap<PolygonMesh, NamedParameters>::const_type
-    fimap = choose_parameter(get_parameter(np, internal_np::face_index),
-                         get_const_property_map(boost::face_index, pmesh));
+  typedef CGAL::Polygon_mesh_processing::Get_index_map_from_NP<boost::face_index_t,
+      CGAL::dynamic_face_property_t<int>,
+      PolygonMesh, NamedParameters, internal_np::face_index_t> MapGetter;
 
+  MapGetter get_map(boost::face_index_t(),
+                    CGAL::dynamic_face_property_t<int>(),
+                    pmesh, np, internal_np::face_index);
+
+  typename MapGetter::PropertyMapType fimap = get_map.property_map();
```

Comment:
you can hide the need of the struct in a function doing the cooking.
If you want to be convenient for the developers (and for the guy that is going to replace all the usage of index maps), you should have something like `typename Default_face_index_map<NP, TM>::const_type fimap = get_initialized_face_index_map(np, pmesh)`. I'm pretty sure a macro can be used to define the 4 possible functions/classes.